src/serde/utils.rs

```diff
@@ -0,0 +1,24 @@
+use std::io;
+use std::io::{copy, sink, Error, Read, Write};
+
+pub fn copy_exactly<R: Read, W: ?Sized + Write>(
+    reader: &mut R,
+    writer: &mut W,
+    expected_size: u64,
+) -> io::Result<()> {
+    let mut reader = reader.by_ref().take(expected_size);
+
+    let count = copy(&mut reader, writer)?;
+    if count < expected_size {
+        Err(Error::new(
+            std::io::ErrorKind::UnexpectedEof,
+            "copy terminated early",
+        ))
+    } else {
+        Ok(())
+    }
+}
+
+pub fn skip_bytes<R: Read>(f: &mut R, skip_size: u64) -> io::Result<()> {
```

Comment:
This seems like a very elaborate (and costly) way of saying
```
f.seek(SeekFrom::Current(skip_size));
```

I think it would be better to just use seek than to actually copy into a sink. If you need this copy function in a future PR, I think you should introduce it there, and with unit tests.

Reply:
The `Read` trait does not have `.seek`.

Reply:
right, you could make `R` `Read + Seek`